app/src/ui/diff/text-diff.tsx

```diff
@@ -747,6 +747,7 @@ export class TextDiff extends React.Component<ITextDiffProps, ITextDiffState> {
         {
           label: this.getDiscardLabel(range.type, range.to - range.from + 1),
           action: () => this.onDiscardChanges(file, range.from, range.to),
+          enabled: !this.props.onHideWhitespaceInDiffChanged,
```

Comment:
I think that's the wrong prop and it disables the context menu option even when whitespace changes are hidden:
```suggestion
          enabled: !this.props.hideWhitespaceInDiff,
```